test/jdk/java/awt/print/PrinterJob/PrintTextTest.java

```diff
@@ -21,193 +21,206 @@
+        book.append(pt, portrait);
+        book.append(pt, landscape);
 
         Font yfont = font.deriveFont(AffineTransform.getScaleInstance(1, 1.5));
-        name = "Page " + new Integer(page++);
-        ptt = new PrintTextTest(name, yfont, null, false);
-        p.add(ptt, BorderLayout.CENTER);
-        p.add(name, ptt);
-        book.append(ptt, portrait);
-        book.append(ptt, landscape);
+        name = "Page " + page++;
+        pt = new PrintText(name, yfont, null, false, preferredSize);
+        pane.addTab(name, pt);
+        book.append(pt, portrait);
+        book.append(pt, landscape);
 
         if (System.getProperty("os.name").startsWith("Windows")) {
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
-            ptt = new PrintJAText(name, font, null, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            name = "Page " + page++;
+            pt = new PrintJapaneseText(name, font, null, true, preferredSize);
+            pane.addTab(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
 
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
+            name = "Page " + page++;
             rotTx = AffineTransform.getRotateInstance(0.15);
-            ptt = new PrintJAText(name, font, rotTx, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            pt = new PrintJapaneseText(name, font, rotTx, true, preferredSize);
+            pane.addTab(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
         }
 
         pjob.setPageable(book);
 
-        JFrame f = new JFrame();
-        f.add(BorderLayout.CENTER, p);
-        f.addWindowListener(new WindowAdapter() {
-            public void windowClosing(WindowEvent e) {System.exit(0);}
+        JButton printButton = new JButton("Print");
+        printButton.addActionListener(event -> {
+            try {
+                if (pjob.printDialog()) {
+                    pjob.print();
+                }
+            } catch (PrinterException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
         });
+
+        JFrame f = new JFrame("PrintTextTest");
+        f.add(BorderLayout.CENTER, pane);
+        f.add(BorderLayout.SOUTH, printButton);
         f.pack();
-        f.show();
 
-        try {
-            if (pjob.printDialog()) {
-                pjob.print();
-            }
-        } catch (PrinterException e) {
-            throw new RuntimeException(e.getMessage());
-        }
+        PassFailJFrame.builder()
+            .title("PrintTextTest")
+            .instructions(INSTRUCTIONS)
```

Comment:
```suggestion
            .instructions(INSTRUCTIONS)
            .testTimeOut(10)
```
I tried running the test and actually comparing the images… the timeout of 5 minutes is too short, we should increase the timeout to 10 or even 15 minutes.

Reply:
Updated.